### ISSUE TITLE
Hour format

### ISSUE THREAD

**Issue by andoruB** [lwindolf/liferea#780]
Just noticed the app uses the 12 hour format, whether or not the system is set to do so. Is there some way that this can be configured differently?

**Comment by rich-coe** [comment]
@andoruB  as an example, what format are you thinking?

**Comment by andoruB** [comment]
This is the de-facto US format, it would be nice if the user could have a choice between different date and hour formats. For example, change between DD-MM-YYYY or MM-DD-YYYY, or YYYY-DD-MM and so on and so forth. And for the hour format, make 24 hour format available (14:00 instead of 2:00PM) also let users choose if the seconds are of relevance (ex. 14:02:20)  
To me it seems this would be a no-brainer, considering most apps and OSes let you choose what hour/date format to use, or use the de-facto one based on the locale selected

**Comment by rich-coe** [comment]
Allow the user to select the date/time format for items in the current year as:
MM/DD HH:MM
DD/MM HH:MM

**Comment by lwindolf** [comment]
@andoruB Liferea uses the locale specific time format. So what you see depends on the locale you have configured. As the user chooses the locale I consider this not to be a bug.

**Comment by andoruB** [comment]
Okay:

`$ cat /etc/default/locale
#  File generated by update-locale
LANG=en_GB.UTF-8
LANGUAGE="en_GB:en"
`

![Screenshot_2021-01-12_23-26-55](https://user-images.githubusercontent.com/4868784/104376516-cad86280-552d-11eb-8443-d70ab90ac65f.png)

Explain please.  
  
Also, what about adding seconds to the time of the post/rss item?

**Comment by lwindolf** [comment]
"Explain please." ?

Could you please explain the expected date format for en_GB.UTF-8 or do you expect me to guess?

Also the relevant environment variable is LC_TIME. LANG and LANGUAGE can be the effective value of LC_TIME but does not need to be. Typing `locale` is the best way to see the effective LC_TIME setting.

About adding seconds: no, I don't think this is useful in the majority of use cases.

**Comment by andoruB** [comment]
```
$ locale
LANG=en_GB.UTF-8
LANGUAGE=en_GB:en
LC_CTYPE="en_GB.UTF-8"
LC_NUMERIC="en_GB.UTF-8"
LC_TIME="en_GB.UTF-8"
LC_COLLATE="en_GB.UTF-8"
LC_MONETARY="en_GB.UTF-8"
LC_MESSAGES="en_GB.UTF-8"
LC_PAPER="en_GB.UTF-8"
LC_NAME="en_GB.UTF-8"
LC_ADDRESS="en_GB.UTF-8"
LC_TELEPHONE="en_GB.UTF-8"
LC_MEASUREMENT="en_GB.UTF-8"
LC_IDENTIFICATION="en_GB.UTF-8"
LC_ALL=
```

For en_GB, in the above screenshot, it should be 5ft January 2021 20:10

**Comment by lwindolf** [comment]
I think you are so fixated on your intended use case, which you are 100% entitled to do so, that you do not see that Liferea implements a completely different one, namely providing the date information in the shortest meaningful manner.

If you really care about the implementation (derived from Evolution) please have a look at the rather complex date format selection logic [here](https://github.com/lwindolf/liferea/blob/8f32f042ec8ba67c50f44c3825ec79c3c77559ea/src/date.c#L111).

**Comment by andoruB** [comment]
That'll output the US standard for time, regardless on which locale I'm on. What's the point on throwing me on the "rabbit hole" of the possibility using the wrong locale, when this is the expected (and hardcoded) output, regardless of what LC_TIME I have set...?

You might think I'm fixated on my use case, but it's not just mine. Not every user of this software lives in the US. It's perfectly reasonable to expect to be able to change the date format, if that's not actually changed according to the locale, as you alluded to earlier.   
  
It's clear that you only want to enable date format in the way you're used to and don't want to give the user any choices in this regard, so I'm done wasting my breath.

**Comment by lwindolf** [comment]
While @andoruB behaviour is not acceptable and I do not care about seeing such behaviour here, he is still right on the localisation issue. Therefore I reopen the ticket

This problem has several layers:
- LC_TIME needs to be followed (works)
- format code needs to be chosen properly (works)
- format code might need to be further localized as we do for some locales already (missing)

This is of course due to apps rarely having an UK/GB translation. Due to missing knowledge I cannot do the translation myself. If someone could contribute it here is how to do it:

1. in po/ copy liferea.pot to en_UK.po (and best also en_GB.po) 
2. fill out header in the new files
3. locate the translations of src/data.c and translate them for proper UK/GB format

Here is the German example which also get's rid of the "%p" causing AM/PM

        #. translation hint: date format for today, reorder format codes as necessary
        #: ../src/date.c:133
        msgid "Today %l:%M %p"
        msgstr "Heute %H:%M"
        
        #. translation hint: date format for yesterday, reorder format codes as necessary
        #: ../src/date.c:142
        msgid "Yesterday %l:%M %p"
        msgstr "Gestern %H:%M"
        
        #. translation hint: date format for dates older than 2 days but not older than a week, reorder format codes as necessary
        #: ../src/date.c:154
        msgid "%a %l:%M %p"
        msgstr "%a %H:%M"
        
        #. translation hint: date format for dates older than a week but from this year, reorder format codes as necessary
        #: ../src/date.c:162
        msgid "%b %d %l:%M %p"
        msgstr "%b %d %H:%M"
        
        #. translation hint: date format for dates from the last years, reorder format codes as necessary
        #: ../src/date.c:165
        msgid "%b %d %Y"
        msgstr "%d %b %Y"

**Comment by sunwire** [comment]
I'm not British of course, but based on the article https://en.wikipedia.org/wiki/Date_and_time_notation_in_the_United_Kingdom
I think it should be like this:

msgstr "Today %H:%M"            e.g. Today 01:23
msgstr "Yesterday %H:%M"      e.g. Yesterday 23:45
msgstr "%a %H:%M"                 e.g. sun 09:34
msgstr "%d %b %H:%M"           e.g. 01 Jan 12:34
msgstr "%d %b %Y"                  e.g. 24 Dec 2021

@andoruB Is this correct?
@lwindolf The file should be named en_GB.po there is no UK country code https://www.iso.org/obp/ui/#iso:code:3166:GB

**Comment by andoruB** [comment]
@sunwire Yes, that would be correct. This is applicable to most european languages/localisations as well to my knowledge. Also, unlike the US localisation, in the UK, the week starts on Monday instead of Sunday, again, applicable to other european locales.

**Comment by lwindolf** [comment]
@sunwire fix will be released with 1.13.6
